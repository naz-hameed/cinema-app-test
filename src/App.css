/* .App {
  text-align: center;
} */

header
{
  background-color: rgb(18, 234, 184);
  color: rgb(0, 54, 80);
  text-align: left;
}

footer
{
  background-color: rgb(18, 234, 184);
  color: rgb(0, 54, 80);
  text-align: center;
}

.signup{
  padding: 20px;
  line-height: 2.5;
}

.signup table {

  width: 50%;
  font-size: 14px;
 
}

.signup button {
  background-color:skyblue;
  border-radius: 6px;
  font-size: 16px;
 
}


.App-logo {
  height: 10vmin;
  pointer-events: none;
  color: blue;
  display: inline;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #646c7a;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: blue;
}

.App-link {
  color: #61dafb;
}

/* @keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
} */
