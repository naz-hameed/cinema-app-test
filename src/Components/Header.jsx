import React from 'react';
import logo from './images/cinema-logo.png';


const Header = () => {
    return (
        <>
        <header>
            <img src={logo} className="App-logo" alt="logo" /> 
            <nav className = "navbar navbar-expand-sm">

            <p>    </p>
            
            <button id = "home-btn">Home</button>
            <button id = "schedule-btn">Schedule</button>
            <button id = "signup-btn">SignUp</button>

           {/* <li> <a href = "#" title ="Home"> Home </a></li> */}
          {/* 
            <li> <a href = "#" title ="Schedule"> Schedule </a></li>
            <li> <a href = "#" title ="Sign Up"> Sign up </a></li> */}
            </nav>
        </header>



        </>
    );
};
export default Header;