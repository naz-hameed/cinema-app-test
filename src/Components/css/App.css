/* .App {
  text-align: center;
} */
header
{
  background-color: white;
  text-align: left;
  margin-bottom: 0px;
}

footer
{
  background-color: white;
  text-align: center;

}

.App-logo {
  height: 10vmin;
  pointer-events: none;
  color: blue;
  display: inline;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-link {
  color: #61dafb;
}

.PageContent {
  background-color: white;
}

.Links {
  list-style-type: none;
}

.form {
  padding-left: 40px;
}


// buttons

.button {
  border: none;
  color: white;
  padding: 16px 32px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
  margin: 4px 2px;
  transition-duration: 0.4s;
  cursor: pointer;
  border-radius: 10px;
}

.centre {
  margin: 0;
  position: absolute;
  left: 50%;
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.menubutton {
  background-color: white; 
  color: black; 
  border: 2px solid #008CBA;
  border-radius: 3px;
}

.menubutton:hover {
  background-color: #008CBA;
  color: white;
}



