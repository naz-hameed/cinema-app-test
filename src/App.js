import React from 'react'
import './App.css';
import Header from './Components/Header';
import Footer from './Components/Footer';
import SignUpPage from './Components/SignUpPage';


function App() {

  return (
    <div className="App">
      <Header /> 
      <SignUpPage />
      <Footer />
    </div>
  );
}

export default App;
