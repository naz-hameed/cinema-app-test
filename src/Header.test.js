import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
configure({ adapter: new Adapter() });
import Header from './Components/Header';
import renderer from 'react-test-renderer';


describe("Testing our header", () => {
    test('Check that there is a home button', () => {
        const testInstance = shallow(<Header />);
        expect(testInstance.find("#home-btn").text()).toBe("Home");
    })

    test('Check that there is a schedule button', () => {
        const testInstance = shallow(<Header />);
        expect(testInstance.find("#schedule-btn").text()).toBe("Schedule");
    })

    test('Check that there is a sign up button', () => {
        const testInstance = shallow(<Header />);
        expect(testInstance.find("#signup-btn").text()).toBe("SignUp");
    })

    test('Check the logo', () => {
        const testInstance = shallow(<Header />);
        expect(testInstance.find("img").prop("src")).toEqual("cinema-logo.png");
    })
});

it('Snapshot test for header rendering correctly', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
})

// test('check the menu', () => {
//     const testInstance = shallow(<Header />);
//     expect(testInstance.find("#Schedule-btn").text()).toBe("Schedule");
//     })

//     test('check the nav bar', () => {
//         const testInstance = shallow(<Header />);
//         expect(testInstance.find("#Navbar").text()).toBe("Navbar");
//         })




