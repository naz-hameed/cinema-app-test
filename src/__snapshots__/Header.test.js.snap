// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for header rendering correctly 1`] = `
<header>
  <br />
  <img
    alt="logo"
    className="App-logo"
    src="skycinemas.png"
  />
</header>
`;
